### PR TITLE
Issue/24 cli options to config

### LINKED ISSUE #24: Move some of the command line options to a config file and add dynamic config loader support
There are some configuration options required by commands that are tied to the environment. To make the work of the plugin easier, an allow storing some of the configuration in VCS we could move them to config file (`airflow-k8s.yml`).

To make the configuration more robust and dynamic we could reuse TemplatedConfigLoader hook. Check how we do it in [kedro-kubeflow-example](https://gitlab.com/getindata/ml-ops/kedro-kubeflow-example/-/blob/master/conf/base/kubeflow.yaml) for reference. 

References from kedro-kubeflow:
* config handler -> [here](https://github.com/getindata/kedro-kubeflow/blob/develop/kedro_kubeflow/config.py)
* dynamic config loader hook -> [here](https://github.com/getindata/kedro-kubeflow/blob/acb9af67c960d8b13cd124ddcc05bf341688e4f6/kedro_kubeflow/hooks.py#L14)
* config loader hook registration -> [here](https://github.com/getindata/kedro-kubeflow/blob/acb9af67c960d8b13cd124ddcc05bf341688e4f6/setup.py#L59)

Before implementing we may have a short discussion which options do we move to config and which of them we allow to override with console call.

### PR DESCRIPTION
**Pull request by michalzelechowski-getindata**
This change makes alignment with kedro-kubeflow regarding configuration, by allowing many of the CLI options to be present in configuration as well. It also allows configuration templating.

Resolves #24

---
Keep in mind: 
- [x] Documentation updates
- [x] [Changelog](CHANGELOG.md) updates

### CODE REVIEW

File: docs/source/02_installation/02_configuration.md [hunk 1]
```diff
@@ -3,12 +3,53 @@
...
+host: {url}
+
+# Directory from where Apache Airflow is reading DAGs definitions
+output: {output}
+
+# Configuration used to run the pipeline
+run_config:
+
+    # Name of the image to run as the pipeline steps
+    image: {image}
```
[the remark this answers is not in the document]
Comment (em-pe) [reply docs/source/02_installation/02_configuration.md:16]: Add section about templated config loader in documentation as well. You can get it from [here](https://kedro-kubeflow.readthedocs.io/en/0.3.0/source/02_installation/02_configuration.html#dynamic-configuration-support) as it works the same way.

File: kedro_airflow_k8s/hooks.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    def read_env() -> Dict:
+        config = ENV_DEFAULTS.copy()
+        overrides = dict(
+            [
+                (k.replace(VAR_PREFIX, "").lower(), v)
+                for k, v in os.environ.copy().items()
+                if k.startswith(VAR_PREFIX)
+            ]
+        )
+        config.update(**overrides)
```
Comment (em-pe) [inline kedro_airflow_k8s/hooks.py:27]: I wonder if it's possible to add a variable here, that we could use in templates to refer to` {package_name}` or `{project name} `?

File: docs/source/02_installation/02_configuration.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (michalzelechowski-getindata) [reply docs/source/02_installation/02_configuration.md:16]: That makes sense, on it.

File: kedro_airflow_k8s/hooks.py (hunk 2 above)
Comment (michalzelechowski-getindata) [reply kedro_airflow_k8s/hooks.py:27]: Variables that are added here are not visible inside DAG template. These have to be injected into templated config files first and then referenced from the DAG.

Comment (em-pe) [reply kedro_airflow_k8s/hooks.py:27]: I was thinking about config templates not DAG templates. What I would want to achieve is to be able to specify in config file:

`experiment_name: {package_name}`

on the other hand package name doesn't change often so it doesn't require templating.

**Review by em-pe — APPROVED** [review]

File: kedro_airflow_k8s/hooks.py (hunk 2 above)
Comment (michalzelechowski-getindata) [reply kedro_airflow_k8s/hooks.py:27]: Yes, it possible to use it this way, given:
`export KEDRO_CONFIG_PACKAGE_NAME=some_package_name`

then in config template:
`experiment_name: ${package_name}`

As you said, for package name probably it's not that required, but this mechanism can be used for any other environment variable with `KEDRO_CONFIG_` prefix.

**Pull request merged**